code/WorkInProgress/telescope/telescopeMisc.dm

```diff
@@ -137,6 +141,47 @@ var/list/magnet_locations = list()
 			var/place = href_list["recieve"]
 			src.lrtrecieve(place)
 
+/obj/machinery/lrteleporter/ui_interact(mob/user, datum/tgui/ui)
+	ui = tgui_process.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "LongRangeTeleporter", name)
+		ui.open()
+	return ui
+
+/obj/machinery/lrteleporter/ui_data(mob/user)
+	var/list/destinations = list()
+	for(var/A in special_places)
+		destinations += list(list(
+			"destination" = "[A]",
+			"ref" = null))
+
+	. = list(
+		"destinations" = destinations
+	)
```

Comment:
Might be missing something, but I don't see where any destination item has a non-null `ref`?

Reply:
You are not. Ref is intended for use with Artemis planetary teleporter, to allow for decoupling from special places.